lib/modules/manager/mise/upgradeable-tooling.ts

```diff
@@ -0,0 +1,148 @@
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { JavaVersionDatasource } from '../../datasource/java-version';
+import { NodeVersionDatasource } from '../../datasource/node-version';
+import { RubyVersionDatasource } from '../../datasource/ruby-version';
+import * as regexVersioning from '../../versioning/regex';
+import * as semverVersioning from '../../versioning/semver';
+import { ToolingConfig, upgradeableTooling } from '../asdf/upgradeable-tooling';
+
+export interface ToolingDefinition {
+  config: ToolingConfig;
+  misePluginUrl?: string;
+}
+
+export const asdfTooling = upgradeableTooling;
+
+export const miseTooling: Record<string, ToolingDefinition> = {
+  bun: {
+    misePluginUrl: 'https://mise.jdx.dev/lang/bun.html',
+    config: {
+      packageName: 'oven-sh/bun',
+      datasource: GithubReleasesDatasource.id,
+      extractVersion: '^bun-v(?<version>\\S+)',
+    },
+  },
+  deno: {
+    misePluginUrl: 'https://mise.jdx.dev/lang/deno.html',
+    config: {
+      packageName: 'denoland/deno',
+      datasource: GithubReleasesDatasource.id,
+      extractVersion: '^v(?<version>\\S+)',
+    },
+  },
+  erlang: {
+    misePluginUrl: 'https://mise.jdx.dev/lang/erlang.html',
+    config: {
+      packageName: 'erlang/otp',
+      datasource: GithubTagsDatasource.id,
+      extractVersion: '^OTP-(?<version>\\S+)',
+      versioning: `${regexVersioning.id}:^(?<major>\\d+?)\\.(?<minor>\\d+?)(\\.(?<patch>\\d+))?$`,
+    },
+  },
+  go: {
+    misePluginUrl: 'https://mise.jdx.dev/lang/go.html',
+    config: {
+      packageName: 'golang/go',
+      datasource: GithubTagsDatasource.id,
+      extractVersion: '^go(?<version>\\S+)',
+    },
+  },
+  java: {
+    misePluginUrl: 'https://mise.jdx.dev/lang/java.html',
+    config: (version) => {
```

Comment:
can we reuse the logic from asdf?

Reply:
Theoretically yes, but there's some issues that I think makes it not worthwhile.

The `asdf` Java plugin does not support all the same types of Java as the `mise` one. i.e. in the `asdf` manager logic, there is no pathways for Zulu Java (`zulu-`), Corretto Java (`corretto-`), or even just a version number (i.e. `java = "17.0.2"`)

- This becomes an issue as `mise` supports having no prefix - defaulting to OpenJDK for no prefix.
- There are plenty of example of these different types of Java / no prefix being used in `.mise.toml` files. You can see on Github Search [here](https://github.com/search?q=mise.toml+java+language%3ATOML&type=code&l=TOML).